frontend/src/features/moti/mapSideBar/containers/MotiInventoryContainer.tsx

```diff
@@ -1,27 +1,113 @@
+  const formikRef = React.useRef<FormikProps<FormikValues>>();
+  const { showSideBar, setShowSideBar, size, context: sidebarContext } = useQueryParamSideBar(
+    formikRef,
+  );
+
+  const latLngSearch = async (latLng?: LatLngLiteral) => {
+    if (!latLng || !latLng?.lat || !latLng?.lng) {
+      toast.error('Unable to perform search, property missing latitude/longitude');
+      return;
+    }
+    try {
+      const parcelLayerResponse = await parcelLayerService.findOneWhereContains(latLng);
+      const properties = getIn(parcelLayerResponse, 'features.0.properties');
+      const ltsaResponse = await getLtsaParcelInfo(properties?.PID);
+      formikParcelDataPopulateCallback(parcelLayerResponse, ltsaResponse?.data);
+    } catch (error) {
+      toast.error('Failed to load parcel info from parcel layer');
+    }
+  };
+
+  const handlePidChange = async (pid: string) => {
+    if (pid) {
+      const ltsaResponse = await getLtsaParcelInfo(pid);
+      try {
+        const parcelLayerResponse = await parcelLayerService.findByPid(pid);
+        formikParcelDataPopulateCallback(parcelLayerResponse, ltsaResponse?.data);
+      } catch (error) {
+        toast.error('Failed to load parcel info from parcel layer');
+      }
+    }
+  };
+
+  const getLtsaParcelInfo = async (pid: string) => {
+    try {
+      return await getParcelInfo(pid);
+    } catch (error) {
+      if (error?.response?.status === 404) {
+        toast.warning(`PID: ${pid} not found in Land Title Direct Search Service.`);
+      } else {
+        toast.error('Failed to load parcel info from Land Title Direct Search Service.');
+      }
+    }
+  };
+
+  /**
+   * Populate the formik form using the passed property data.
+   * @param nameSpace the formik namespace that should be used to write any retrieved data.
+   * @param matchingParcel the parcel to use to populate the formik form.
+   */
+  const formikParcelDataPopulateCallback = (
+    parcelLayerResponse: FeatureCollection<Geometry, GeoJsonProperties>,
+    ltsaResponse: ParcelInfo | undefined,
+    nameSpace?: string,
+  ) => {
+    //TODO: populate form with response data for PSP-705
+  };
 
+  //hooks
+  const { getParcelInfo } = useApiLtsa();
+  const parcelLayerService = useLayerQuery(PARCELS_LAYER_URL);
+  const { setMovingPinNameSpace } = useMarkerSearch(formikRef, showSideBar, latLngSearch);
   return (
     <MapSideBarLayout
-      title="Add to Inventory"
+      title="Add Titled Property to Inventory"
       show={showSideBar}
       setShowSideBar={setShowSideBar}
       size={size}
       hidePolicy={true}
     >
-      <SubmitPropertySelector addProperty={noop} />
+      {!sidebarContext || sidebarContext === SidebarContextType.ADD_PROPERTY_TYPE_SELECTOR ? (
+        <SubmitPropertySelector
+          addProperty={() => setShowSideBar(true, SidebarContextType.ADD_BARE_LAND, 'wide', true)}
+        />
+      ) : (
+        <>
+          <PropertySearchForm
```

Comment:
The PropertySearchForm provides two features, one for when PID changes, the other when Civic Address changes?  Makes me think there may be a more generic implementation that will support all different types of changes.  Then let the container act on the change based on the type.

Reply:
What you are talking about is a common react pattern, where you replace x function event handlers with a single reducer based event handler. Personally I think that pattern is better used where x >=3 due to the overhead involved.